app/src/main/java/com/example/sharewhatyoucanproject/login/LoginViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package com.example.sharewhatyoucanproject.login
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.DEFAULT_PASSWORD
+import com.example.sharewhatyoucanproject.utils.getRandomName
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.UserProfileChangeRequest
+import com.google.firebase.firestore.FirebaseFirestore
+
+class LoginViewModel(
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance(),
+    private val db: FirebaseFirestore = FirebaseFirestore.getInstance(),
+
+) : ViewModel() {
+
+    val checkResult = MutableLiveData<String>()
```

Comment:
only the view model should be able to modify its live data. A common pattern is to create a private instance of mutable live data and a public instance of live data that maps to the mutable live data.

```
private val _checkResult = MutableLiveData<String>()
val checkResult: LiveData<String> = _checkResult
```